platformio/commands/debug/command.py

```diff
@@ -166,10 +166,22 @@ def cli(ctx, project_dir, project_conf, environment, verbose, interface, __unpro
     # pylint: disable=import-outside-toplevel
     from platformio.commands.debug.process.client import GDBClient, reactor
 
-    client = GDBClient(project_dir, __unprocessed, debug_options, env_options)
-    client.spawn(ide_data["gdb_path"], ide_data["prog_path"])
+    # pylint: disable=no-member
+    GDBClient(project_dir, __unprocessed, debug_options, env_options) \
+    .spawn(ide_data["gdb_path"], ide_data["prog_path"]) \
+    .addErrback(handleFailure)
```

Comment:
This new errback catches exception `DebugInvalidOptionsError` asynchronously raised by `DebugServer::spawn()`
when `server_executable` cannot be found. The exception was being suppressed because (a) there was no errback and (b) as a result of being raised from a Deferred, the exception couldn't bubble up into the main `try...except` handler in `platformio/__main__.py`.